dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java

```diff
@@ -14,122 +18,135 @@
+            "object to the specified environment(s). Multiple environments can be separated by a comma (',')";
+    private static final String PARAM_ENVIRONMENT = "environment";
+    private static final String PARAM_FORCE_PUSH = "force";
+
     private PublisherAPI publisherAPI = PublisherAPI.getInstance();
+    private EnvironmentAPI environmentAPI = APILocator.getEnvironmentAPI();
+    private BundleAPI bundleAPI = APILocator.getBundleAPI();
+    private UserAPI userAPI = APILocator.getUserAPI();
+    private RoleAPI roleAPI = APILocator.getRoleAPI();
 
     @Override
     public List<WorkflowActionletParameter> getParameters() {
-        List<WorkflowActionletParameter> params = new ArrayList<WorkflowActionletParameter>();
-
-        params.add(new WorkflowActionletParameter("environment", "Name of the Enviroment", "", true));
-        params.add(new WorkflowActionletParameter("force", "Force the Push? true or false", "false", true));
-
+        final List<WorkflowActionletParameter> params = new ArrayList<>();
+        params.add(new WorkflowActionletParameter(PARAM_ENVIRONMENT, "Name of the Environment", "", true));
+        params.add(new WorkflowActionletParameter(PARAM_FORCE_PUSH, "Force the Push? true or false", "false", true));
         return params;
     }
 
     @Override
     public String getName() {
-        return "Push Now";
+        return ACTIONLET_NAME;
     }
 
     @Override
     public String getHowTo() {
-        return "This actionlet will automatically publish the the content object to the specified enviroment(s). Multiple Environments can be separated by a comma";
+        return ACTIONLET_DESCRIPTION;
     }
 
-    public void executeAction(WorkflowProcessor processor, Map<String, WorkflowActionClassParameter> params)
+    @Override
+    public void executeAction(final WorkflowProcessor processor, final Map<String, WorkflowActionClassParameter> params)
             throws WorkflowActionFailureException {
+        final List<String> identifiers = new ArrayList<>();
+        final Contentlet contentlet = processor.getContentlet();
+        final User user = processor.getUser();
+        final String environments = params.get(PARAM_ENVIRONMENT).getValue();
         try {
-            // Gets available languages
-            // List<Language> languages = languagesAPI.getLanguages();
-
-            Contentlet ref = processor.getContentlet();
-
-            boolean _contentPushNeverExpire = true;
-
-            boolean forcePush = ("true".equals(params.get("force").getValue())) ? true : false;
-            String envrions = params.get("environment").getValue();
-            if (envrions == null) {
-                Logger.error(this.getClass(), "There are no environments set to push to");
+            final boolean forcePush = "true".equals(params.get(PARAM_FORCE_PUSH).getValue()) ? Boolean.TRUE : Boolean.FALSE;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)